evap/staff/tests/test_views.py

```diff
@@ -2229,16 +2230,33 @@ def test_questionnaire_with_answers_warning(self):
             '<label class="form-check-label badge bg-danger" for="id_contributions-1-questionnaires_0">', page
         )
 
-    @patch.dict(Evaluation.STATE_STR_CONVERSION, {Evaluation.State.PREPARED: "mock-translated-prepared"})
     def test_state_change_log_translated(self):
-        page = self.app.get(self.url, user=self.manager)
-        self.assertNotIn("mock-translated-prepared", page)
-
         self.evaluation.ready_for_editors()
         self.evaluation.save()
 
-        page = self.app.get(self.url, user=self.manager)
-        self.assertIn("mock-translated-prepared", page)
+        translation.activate("en")
+        self.manager.language = "en"
+        self.manager.save()
+
+        page_en = self.app.get(self.url, user=self.manager)
+
+        new_en = translation.gettext("new")
+        prepared_en = translation.gettext("prepared")
```

Comment:
I think we can lean a bit more towards integration-testing and use "neu" etc. directly

Reply:
I'm not sure if we've ever done that (make tests rely on the translation stored in messages.po). For some other tests, I think we mock `gettext` or other translation functions to make them return some unique string. The tests then
* look for the existence of the unique string on the page
* assert that the mock was called with the right input

This might help with https://github.com/e-valuation/EvaP/pull/2174#discussion_r1585330486 above, but it might also be overkill. If using the translations stored in messages.mo helps to keep it simple here, that's also fine with me

Reply:
nevermind. It seems like this mocking persists past the end of the test.

Reply:
I could not get translation with mocking to work and decided to just match for the full html fragment instead (like you @niklasmohrin thought, the response of course contained the string "new" :) )